histomicstk/annotations_and_masks/annotations_to_masks_handler.py

```diff
@@ -0,0 +1,284 @@
+        relIdxs = []
+        for group_name in relevant_groups:
+            relIdxs.extend(list(elinfos_roi.loc[
+                elinfos_roi.group == group_name, :].index))
+
+        # get relevnt infos and sort from largest to smallest (by bbox area)
+        # so that the smaller elements are layered last. This helps partially
+        # address issues describe in:
+        # https://github.com/DigitalSlideArchive/HistomicsTK/issues/675
+        elinfos_relevant = elinfos_roi.loc[relIdxs, :].copy()
+        elinfos_relevant.sort_values(
+            'bbox_area', axis=0, ascending=False, inplace=True)
+
+        # Go through elements and add to ROI mask
+        for elId, elinfo in elinfos_relevant.iterrows():
+
+            elNo += 1
+            elcountStr = "%s: Overlay level %d: Element %d of %d: %s" % (
+                monitorPrefix, overlay_level, elNo, N_elements, elinfo['group'])
+            if verbose:
+                print(elcountStr)
+
+            # now add element to ROI
+            ROI = _get_and_add_element_to_roi(
+                elinfo=elinfo, slide_annotations=slide_annotations, ROI=ROI,
+                roiinfo=roiinfo, roi_polygon=roi_polygon,
+                GT_code=GTCodes_df.loc[elinfo['group'], 'GT_code'],
+                use_shapely=use_shapely, verbose=verbose,
+                monitorPrefix=elcountStr)
+
+            # save a copy of ROI-only mask to crop to it later if needed
+            if crop_to_roi and (overlay_level == GTCodes_df.loc[
+                    roi_group, 'overlay_order']):
+                roi_only_mask = ROI.copy()
+
+    # Crop polygons to roi if needed (prevent 'overflow' beyond roi edge)
+    if crop_to_roi:
+        ROI[roi_only_mask == 0] = 0
+
+    # tighten boundary --remember, so far we've use element bboxes to
+    # make an over-estimated margin around ROI boundary.
+    nz = np.nonzero(ROI)
+    ymin, xmin = [np.min(arr) for arr in nz]
+    ymax, xmax = [np.max(arr) for arr in nz]
+    ROI = ROI[ymin:ymax, xmin:xmax]
+
+    # update roi offset
+    roiinfo['XMIN'] += xmin
+    roiinfo['YMIN'] += ymin
+    roiinfo['XMAX'] += xmin
+    roiinfo['YMAX'] += ymin
+    roiinfo['BBOX_WIDTH'] = roiinfo['XMAX'] - roiinfo['XMIN']
+    roiinfo['BBOX_HEIGHT'] = roiinfo['YMAX'] - roiinfo['YMIN']
+
+    return ROI, roiinfo
+
+# %% =====================================================================
+
+
+def get_all_roi_masks_for_slide(
+        gc, slide_id, GTCODE_PATH, MASK_SAVEPATH, slide_name=None,
+        verbose=True, monitorPrefix="", get_roi_mask_kwargs=dict()):
+    """Parse annotations and saves ground truth masks for ALL ROIs.
+
+    Get all ROIs in a single slide. This is a wrapper around get_roi_mask()
+    which should be referred to for implementation details.
+
+    Parameters
+    -----------
+    gc : object
+        girder client object to make requests, for example:
+        gc = girder_client.GirderClient(apiUrl = APIURL)
```

Comment:
In our other tests, we add files that are to big to include in the repo as a sha512 hash that gets downloaded from data.kitware.com (listed in plugin_tests/data).  `plugin.cmake` lists this as external data, and then the test "upload" the file to a test instance of a girder server.  This provides a bit more isolation in the tests (and reduces network traffic, as the files aren't redownloaded if they have already been obtained).

I can refactor this to use that method, if you like.

Reply:
@manthey Yes please, that would be great!! Thank you!